sdk/internal/testframework/recording.go

```diff
@@ -0,0 +1,416 @@
+	recorder                 *recorder.Recorder
+	src                      rand.Source
+	now                      *time.Time
+	Sanitizer                *RecordingSanitizer
+	Matcher                  *RequestMatcher
+	c                        TestContext
+}
+
+const (
+	alphanumericBytes           = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890"
+	alphanumericLowercaseBytes  = "abcdefghijklmnopqrstuvwxyz1234567890"
+	randomSeedVariableName      = "randomSeed"
+	nowVariableName             = "now"
+	ModeEnvironmentVariableName = "AZURE_TEST_MODE"
+)
+
+// Inspired by https://stackoverflow.com/questions/22892120/how-to-generate-a-random-string-of-a-fixed-length-in-go
+const (
+	letterIdxBits = 6                    // 6 bits to represent a letter index
+	letterIdxMask = 1<<letterIdxBits - 1 // All 1-bits, as many as letterIdxBits
+	letterIdxMax  = 63 / letterIdxBits   // # of letter indices fitting in 63 bits
+)
+
+type RecordMode string
+
+const (
+	Record   RecordMode = "record"
+	Playback RecordMode = "playback"
+	Live     RecordMode = "live"
+)
+
+type VariableType string
+
+const (
+	Default             VariableType = "default"
+	Secret_String       VariableType = "secret_string"
+	Secret_Base64String VariableType = "secret_base64String"
+)
+
+// NewRecording initializes a new Recording instance
+func NewRecording(c TestContext, mode RecordMode) (*Recording, error) {
+	// create recorder based on the test name, recordMode, variables, and sanitizers
+	recPath, varPath := getFilePaths(c.Name())
+	rec, err := recorder.NewAsMode(recPath, modeMap[mode], nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the mode is set in the environment, let that override the requested mode
+	// This is to enable support for nightly live test pipelines
+	envMode := getOptionalEnv(ModeEnvironmentVariableName, string(mode))
+	mode = RecordMode(*envMode)
+
+	// initialize the Recording
+	recording := &Recording{
+		SessionName:              recPath,
+		RecordingFile:            recPath + ".yaml",
+		VariablesFile:            varPath,
+		Mode:                     mode,
+		variables:                make(map[string]*string),
+		previousSessionVariables: make(map[string]*string),
+		recorder:                 rec,
+		c:                        c,
+	}
+
+	// Try loading the recording if it already exists to hydrate the variables
+	err = recording.initVariables()
+	if err != nil {
+		return nil, err
+	}
+
+	// set the recorder Matcher
+	recording.Matcher = DefaultMatcher(c)
+	rec.SetMatcher(recording.matchRequest)
+
+	// wire up the sanitizer
+	recording.Sanitizer = DefaultSanitizer(rec)
+
+	return recording, err
+}
+
+// GetRecordedVariable returns a recorded variable. If the variable is not found we return an error
+// variableType determines how the recorded variable will be saved. Default indicates that the value should be saved without any sanitation.
+func (r *Recording) GetRecordedVariable(name string, variableType VariableType) (string, error) {
```

Comment:
I think this should be renamed to something like `GetEnvVar()` and the doc comment updated appropriately.  It wasn't clear to me that this retrieved the value from the environment until I dug into the implementation.